root/static/scripts/edit/URLCleanup.js

```diff
@@ -1747,6 +1747,22 @@ const CLEANUPS = {
       return /^https:\/\/myspace\.com\//.test(url);
     },
   },
+  'napster': {
+    match: [new RegExp('^(https?://)?[a-z]{2}\\.napster\\.com', 'i')],
+    type: LINK_TYPES.streamingpaid,
+    clean: function (url) {
+      return url.replace(/^http:\/\//, 'https://');
+    },
+    validate: function (url, id) {
+      switch (id) {
+        case LINK_TYPES.streamingpaid.artist:
+          return /^https:\/\/[a-z]{2}\.napster\.com\/artist\/[^\/]+$/.test(url);
+        case LINK_TYPES.streamingpaid.release:
+          return /^https:\/\/[a-z]{2}\.napster\.com\/artist\/[^\/]+\/album\/[^\/]+$/.test(url);
```

Comment:
Validation of URL ending makes sense, but it should probably be accompanied with the removal of superfluous trailing characters in the function `clean`.

Validation could be improved to discard unneeded query`?…` and fragment `#…` too.

Reply:
Do you have example URLs with those? I haven't seen any trailing characters being added while I clicked around.

Reply:
* I could not find an actual example with query, so there is no reason to allow it as it can be used to include irrelevant referral or whatever: https://fr.napster.com/artist/bread?ref=spammer
* Most actions in the page have an empty fragment: https://fr.napster.com/artist/bread#
* Also found out that track’s link is a subpath of album: https://fr.napster.com/artist/various-artists/album/70-hits-of-the-70s/track/guitar-man

Reply:
I knew about the track, but track isn't the same as recording, so I wasn't sure whether to allow linking to it or not. That said, I guess we do allow it for Spotify...

Reply:
Ok, made those three changes (remove # and ?, allow track links on recordings).